src/main/java/dev/openfeature/sdk/ProviderRepository.java

```diff
@@ -1,24 +1,25 @@
 package dev.openfeature.sdk;
 
-import lombok.extern.slf4j.Slf4j;
-
 import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.atomic.AtomicReference;
-import java.util.function.Consumer;
 import java.util.stream.Stream;
 
+import javax.annotation.Nullable;
+
+import lombok.extern.slf4j.Slf4j;
+
 @Slf4j
 class ProviderRepository {
 
     private final Map<String, FeatureProvider> providers = new ConcurrentHashMap<>();
-    private final ExecutorService taskExecutor = Executors.newCachedThreadPool();
-    private final Map<String, FeatureProvider> initializingNamedProviders = new ConcurrentHashMap<>();
+    private final Map<String, EventEmitter> clientEmitters = new ConcurrentHashMap<>();
```

Comment:
This map get populated for client names, but we never clean (`Map.remove()`) entries from it 🤔  I think we need a way to clean up this

Reply:
hmmm I guess that would be done if we "unset" a client? I'll think about that.

Reply:
Yeah, I think we must not define the `unset` behavior, but detect garbage collection of a client and then perform cleanup, or any other mechanism.

If we are tying this to GC, then we should be able to rely on Object.finalize() [1]

[1] - https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#finalize--

Reply:
Thanks to our discussion on this subject, I am convinced that this is not a blocker nor requires further investigation. What we should do is enhance readme on the point that client event handlers (scoped event handlers) require manual handler removal if they are intended to be dynamically created (ex:- creating a unique client per each request)

Reply:
I've made changes to only allocate memory for clients that actually add handlers.